server/container/util-java8/src/main/java/org/apache/james/util/FluentFutureStream.java

```diff
@@ -33,6 +34,16 @@
         return new FluentFutureStream<>(completableFuture);
     }
 
+    public static <T> FluentFutureStream<T> ofUnboxedStreams(Stream<CompletableFuture<Stream<T>>> completableFuture) {
```

Comment:
Naming is not clear for me what do you mean by "unboxed" ?
I think "ofNestedStreams" would have been better.